### ISSUE TITLE
"TypeError: Rx.Observable.ofObjectChanges is not a function"

### ISSUE THREAD

**Issue by hungdluit** [Reactive-Extensions/RxJS#1147]
I'm using typescript version.
tried to create observer object from a class. But get the error "**TypeError: Rx.Observable.ofObjectChanges is not a function**"

`"use strict";

/// <reference path="node_modules/rx/ts/rx.all.es6.d.ts" />

import Rx = require("rx");

class tester {
    public name: string;
}

var t = new tester();
var source = Rx.Observable.ofObjectChanges(t);

var subscription = source.subscribe(
    function (x) {
        console.log('Next: %s', x);
    },
    function (err) {
        console.log('Error: %s', err);
    },
    function () {
        console.log('Completed');
    });

t.name = "hung";

I tried to take some research but cannot find the solution.
Can someone here help:)

**Comment by basarat** [comment]
I investigated this. Its because `rx` points to `index.js` in node builds.

`package.json` has:

```
"main": "index.js",

"browser": {
    "index.js": "dist/rx.all.js"
  },
```

So if a user does `import Rx = require('rx')` they don't get all the functions in a node environment 

# Workaround 

```ts
/// <reference path="./node_modules/rx/ts/rx.all.d.ts"/>

import rx = require("rx");

// Fix rx to point to rx.all
var Rx = rx;
declare var require:any; // not needed if you have node.d.ts
Rx = require('rx/dist/rx.all');

console.log(Rx.Observable.ofObjectChanges);
```

:rose:

**Comment by palra** [comment]
For the moment, `index.js` requires each RxJs modules, one per one. Why is the `main` field in the `package.json` can't just be `dist/rx.all.js` ? Isn't the `rx.all.js` supposed to include every RxJs submodules ?

**Comment by basarat** [comment]
@palra I agree with your solution. Having a different path for `node` vs. `browser` doesn't seem needed :rose:

**Comment by atom2ueki** [comment]
@hungdluit this function no longer works in the latest rxjs 5, refer to this https://github.com/ReactiveX/rxjs/blob/master/MIGRATION.md
